symengine/functions.cpp

```diff
@@ -3571,4 +3620,23 @@ RCP<const Basic> min(const vec_basic &arg)
     }
 }
 
+RCP<const Basic> expand_as_exp(const RCP<const Basic> &self)
+{
+    map_basic_basic mp_args;
+    for (const auto &a : self->get_args()) {
+        mp_args[a] = expand_as_exp(a);
+    }
```

Comment:
Change `bottom_up` to a `Visitor` class and extend that for `expand_as_exp` by overriding the `void bvisit(const Sin &)` method and other methods.